lib/ruby_indexer/lib/ruby_indexer/index.rb

```diff
@@ -0,0 +1,113 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  class Index
+    extend T::Sig
+
+    sig { returns(T::Hash[String, T::Array[Entry]]) }
+    attr_reader :entries
+
+    sig { void }
+    def initialize
+      # Holds all entries in the index using the following format:
+      # {
+      #  "Foo" => [#<Entry::Class>, #<Entry::Class>],
+      #  "Foo::Bar" => [#<Entry::Class>],
+      # }
+      @entries = T.let({}, T::Hash[String, T::Array[Entry]])
+
+      # Holds references to where entries where discovered so that we can easily delete them
+      # {
+      #  "/my/project/foo.rb" => ["Foo"],
+      #  "/my/project/bar.rb" => ["Foo::Bar"],
+      # }
+      @files_to_entries = T.let({}, T::Hash[String, T::Array[String]])
```

Comment:
Should this be pointing to the `entries` rather than namespaces? This would avoid the string comparison to delete each entry?

Reply:
If we had entries instead of names, how would we find the right entry to remove? We currently, take all entries for a particular class/module
```ruby
@entries[name]
```
And then remove the ones that were pushed by that specific file path.

Maybe I'm not seeing your idea.

Reply:
By comparing instances rather than strings:

```rb
my_entry = Class.new
@entries["Foo"] << my_entry
@entries_by_path["foo.rb"] << my_entry

@entries["Foo"].last.equal?(@entries_by_path["foo.rb"].last) # => true
```